mesonbuild/scripts/scanbuild.py

```diff
@@ -37,6 +38,13 @@ def run(args):
     logdir = os.path.join(blddir, 'meson-logs/scanbuild')
     shutil.rmtree(logdir, ignore_errors=True)
 
+    cmd = coredata.get_cmd_line_file(blddir)
+    data = coredata.CmdLineFileParser()
+    data.read(cmd)
+
+    if 'cross_file' in data['properties'].keys():
+        meson_cmd.append('--cross-file={}'.format(literal_eval(data['properties']['cross_file'])[0]))
+
```

Comment:
This is wrong too, this should be something like:
```python
if 'cross_file' in data['properties']:  # note: not .keys()
    meson_cmd.extend(['-cross-file={}'.format(f) for f in literal_eval(data['properties']['cross_file']))])
```

And the same for native files. You need to load all of the cross (and native files), not just the first one.